bubbles-map/src/LatLong.tsx

```diff
@@ -1,76 +1,102 @@
 import React, { Component } from "react";
-import {
-  Map,
-  CircleMarker,
-  TileLayer,
-  Tooltip,
-  Marker,
-  Popup,
-} from "react-leaflet";
+import { Data } from "./Data";
+import { Map, CircleMarker, TileLayer, Tooltip, Viewport } from "react-leaflet";
 import "leaflet/dist/leaflet.css";
-import data from "./cities";
 
 export interface LatLongProps {
-  latitude?: number;
-  longitude?: number;
-  titleenabled?: boolean;
-  title?: string;
-  zoom?: number;
-  bubblecolor?: string;
+  latitude: number;
+  longitude: number;
+  titleenabled: boolean;
+  title: string;
+  zoom: number;
+  bubblecolor: string;
+  data: Data[];
+  maxRadius: number;
+  minRadius: number;
+  resizeBubbles: boolean;
 }
 
-interface State {
-    latitude?: number;
-    longitude?: number;
-    titleenabled?: boolean;
-    title?: string;
-    zoom?: number;
-    bubblecolor?: string;
+export interface LatLongState {
+  currentZoom: number;
 }
 
-export class LatLong extends Component<LatLongProps, State> {
-  renderTitle(){
-    if(this.props.titleenabled==true){
-      return <h3 style={{ textAlign: "center" }}>{this.props.title || "Most Populous Cities in Asia"}</h3>;
-    }else{
+function map_range(value, low1, high1, low2, high2) {
```

Comment:
Is this to maintain the scale?

Reply:
Yes, it should find the corresponding radius value for a given dataset value